pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -107,6 +115,26 @@ public ProducerImpl(PulsarClientImpl client, String topic, String producerName,
         this.compressor = CompressionCodecProvider
                 .getCompressionCodec(convertCompressionType(conf.getCompressionType()));
 
+        if (conf.isEncryptionEnabled()) {
+            this.msgCrypto = new MessageCrypto(true);
+
+            // Regenerate data key cipher at fixed interval
+            keyGenExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
it seems we are not closing this executor.??

Reply:
Addressed.